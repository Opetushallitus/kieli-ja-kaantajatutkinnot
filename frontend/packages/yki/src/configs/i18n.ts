import i18n, { changeLanguage, t, use } from 'i18next';
import LanguageDetector from 'i18next-browser-languagedetector';
import {
  initReactI18next,
  useTranslation,
  UseTranslationOptions,
} from 'react-i18next';
import { AppLanguage, I18nNamespace } from 'shared/enums';
import { DateUtils } from 'shared/utils';

import commonEN from 'public/i18n/en-GB/common.json';
import publicEN from 'public/i18n/en-GB/public.json';
import commonFI from 'public/i18n/fi-FI/common.json';
import publicFI from 'public/i18n/fi-FI/public.json';
import commonSV from 'public/i18n/sv-SE/common.json';
import publicSV from 'public/i18n/sv-SE/public.json';

// Defaults and resources
const langFI = AppLanguage.Finnish;
const langSV = AppLanguage.Swedish;
const langEN = AppLanguage.English;

const supportedLangs = [langFI, langSV, langEN];

const resources = {
  [langFI]: {
    [I18nNamespace.Common]: commonFI,
    [I18nNamespace.Public]: publicFI,
  },
  [langSV]: {
    [I18nNamespace.Common]: commonSV,
    [I18nNamespace.Public]: publicSV,
  },
  [langEN]: {
    [I18nNamespace.Common]: commonEN,
    [I18nNamespace.Public]: publicEN,
  },
};

const detectionOptions = {
  order: ['localStorage', 'htmlTag'],
  caches: ['localStorage'],
  lookupLocalStorage: 'i18NextLng_yki',
};

// TypeScript definitions for react-i18next. IDE might show this to be unused, but ts does some type checking with it.
declare module 'react-i18next' {
  interface CustomTypeOptions {
    defaultNS: typeof langFI;
    resources: {
      [langFI]: typeof commonFI;
      [langSV]: typeof commonSV;
      [langEN]: typeof commonEN;
    };
  }
}

export const initI18n = () => {
  const i18n = use(initReactI18next).use(LanguageDetector).init({
    defaultNS: 'common',
    resources,
    detection: detectionOptions,
    fallbackLng: langFI,
    load: 'currentOnly',
    debug: !REACT_ENV_PRODUCTION,
  });
  const currentLanguage = getCurrentLang() as AppLanguage;
  DateUtils.setDayjsLocale(currentLanguage);

  return i18n;
};

const useAppTranslation = (
  options: UseTranslationOptions<string>,
  ns: I18nNamespace
) => {
  // @ts-expect-error ts import fail
  return useTranslation(ns, options);
};

export const usePublicTranslation = (
  options: UseTranslationOptions<string>
) => {
  return useAppTranslation(options, I18nNamespace.Public);
};

export const useCommonTranslation = () => {
  const { t } = useAppTranslation(
    {
      keyPrefix: 'yki.common',
    },
    I18nNamespace.Common
  );

  return t;
};

// ts-unused-exports:disable-next-line
export const translateOutsideComponent = () => {
  return t;
};

export const getCurrentLang = (): AppLanguage => {
  return i18n.language as AppLanguage;
};

export const getSupportedLangs = (): Array<AppLanguage> => {
  return supportedLangs;
};

export const changeLang = (language: AppLanguage) => {
  DateUtils.setDayjsLocale(language);

  return changeLanguage(language);
};
