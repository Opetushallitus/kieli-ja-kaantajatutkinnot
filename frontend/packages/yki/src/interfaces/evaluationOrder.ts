export interface ExaminationParts {
  readingComprehension: boolean;
  speechComprehension: boolean;
  speaking: boolean;
  writing: boolean;
}

export interface ParticipantDetails {
  firstNames?: string;
  lastName?: string;
  email?: string;
  birthdate?: string;
}

export type Subtest = 'LISTENING' | 'READING' | 'SPEAKING' | 'WRITING';

export interface EvaluationOrderRequest {
  first_names: string;
  last_name: string;
  email: string;
  birthdate: string;
  subtests: Array<Subtest>;
}

export interface EvaluationOrderResponse {
  redirect: string;
}
