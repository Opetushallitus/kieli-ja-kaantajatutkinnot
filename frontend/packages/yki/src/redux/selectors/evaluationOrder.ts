import { RootState } from 'configs/redux';

export const evaluationOrderSelector = (state: RootState) =>
  state.evaluationOrder;
