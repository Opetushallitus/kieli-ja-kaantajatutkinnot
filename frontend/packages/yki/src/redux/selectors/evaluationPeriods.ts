import { RootState } from 'configs/redux';

export const evaluationPeriodsSelector = (state: RootState) =>
  state.evaluationPeriods;
