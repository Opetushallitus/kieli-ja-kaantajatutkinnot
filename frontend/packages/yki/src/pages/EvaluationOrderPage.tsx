import { Box, Grid } from '@mui/material';
import { useCallback, useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { H1, H2, HeaderSeparator } from 'shared/components';
import { APIResponseStatus, Severity } from 'shared/enums';
import { useToast } from 'shared/hooks';

import { PublicEvaluationOrderForm } from 'components/reassessment/evaluationOrder/PublicEvaluationOrderForm';
import { PublicEvaluationOrderPageSkeleton } from 'components/skeletons/PublicEvaluationOrderPageSkeleton';
import { usePublicTranslation } from 'configs/i18n';
import { useAppDispatch, useAppSelector } from 'configs/redux';
import { useNavigationProtection } from 'hooks/useNavigationProtection';
import {
  initialState,
  loadEvaluationPeriod,
  resetEvaluationOrderState,
} from 'redux/reducers/evaluationOrder';
import { evaluationOrderSelector } from 'redux/selectors/evaluationOrder';

const EvaluationOrderPage = () => {
  // i18n
  const { t } = usePublicTranslation({
    keyPrefix: 'yki.pages.evaluationOrderPage',
  });

  const { showToast } = useToast();

  // Redux
  const dispatch = useAppDispatch();
  const {
    acceptConditions,
    loadPeriodState,
    submitOrderState,
    evaluationPeriod,
    participantDetails,
    examinationParts,
    evaluationPaymentRedirect,
  } = useAppSelector(evaluationOrderSelector);

  // React Router
  const params = useParams();

  // Navigation protection
  // Enable on changed form details, disable when redirecting to payments provider.
  const isDirtyState =
    initialState.acceptConditions !== acceptConditions ||
    initialState.participantDetails !== participantDetails ||
    initialState.examinationParts !== examinationParts;
  useNavigationProtection(isDirtyState && !evaluationPaymentRedirect);

  // Reset user filled details when unmounting (eg. navigating away from page)
  const resetStateOnUnmount = useCallback(() => {
    dispatch(resetEvaluationOrderState());
  }, [dispatch]);
  useEffect(() => resetStateOnUnmount, [resetStateOnUnmount]);

  useEffect(() => {
    if (loadPeriodState === APIResponseStatus.Error) {
      showToast({
        severity: Severity.Error,
        description: t('toasts.loadingError'),
      });
    } else if (
      loadPeriodState === APIResponseStatus.NotStarted &&
      !evaluationPeriod?.id &&
      params.evaluationId
    ) {
      // Fetch evaluation details
      dispatch(loadEvaluationPeriod(+params.evaluationId));
    } else if (
      submitOrderState === APIResponseStatus.Success &&
      evaluationPaymentRedirect
    ) {
      window.location.replace(evaluationPaymentRedirect);
    }
  }, [
    dispatch,
    showToast,
    t,
    params.evaluationId,
    evaluationPeriod?.id,
    loadPeriodState,
    submitOrderState,
    evaluationPaymentRedirect,
  ]);

  switch (loadPeriodState) {
    case APIResponseStatus.NotStarted:
    case APIResponseStatus.InProgress:
      return <PublicEvaluationOrderPageSkeleton />;
    case APIResponseStatus.Cancelled:
    case APIResponseStatus.Error:
      return (
        <Box className="public-evaluation-order-page">
          <Grid
            container
            rowSpacing={4}
            direction="column"
            className="public-evaluation-order-page__grid-container"
          >
            <Grid
              item
              className="public-evaluation-order-page__grid-container__item-header"
            >
              <H1 className="public-evaluation-order-page__title-heading">
                {t('title')}
              </H1>
              <HeaderSeparator />
              <H2>{t('notFound')}</H2>
            </Grid>
          </Grid>
        </Box>
      );
    case APIResponseStatus.Success:
      return (
        <Box className="public-evaluation-order-page">
          <Grid
            container
            rowSpacing={4}
            direction="column"
            className="public-evaluation-order-page__grid-container"
          >
            <Grid
              item
              className="public-evaluation-order-page__grid-container__item-header"
            >
              <H1 className="public-evaluation-order-page__title-heading">
                {t('title')}
              </H1>
              <HeaderSeparator />
            </Grid>
            <PublicEvaluationOrderForm />
          </Grid>
        </Box>
      );
  }
};

export default EvaluationOrderPage;
