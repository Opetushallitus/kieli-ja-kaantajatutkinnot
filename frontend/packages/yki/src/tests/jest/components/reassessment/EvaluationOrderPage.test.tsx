import renderer from 'react-test-renderer';
import { APIResponseStatus } from 'shared/enums';

import { initI18nForTests } from 'configs/i18n';
import { EvaluationOrderPage } from 'pages/EvaluationOrderPage';
import { initialState as initialEvaluationOrderState } from 'redux/reducers/evaluationOrder';
import { DefaultProviders } from 'tests/jest/utils/DefaultProviders';
import { renderWithProviders } from 'tests/jest/utils/store';
import { evaluationPeriods } from 'tests/msw/fixtures/evaluationPeriods';
import { SerializationUtils } from 'utils/serialization';

jest.unmock('configs/i18n');
jest.requireActual('configs/i18n');

beforeAll(() => {
  initI18nForTests();
});

describe('EvaluationOrderPage', () => {
  const preloadedState = {
    evaluationOrder: {
      ...initialEvaluationOrderState,
      loadPeriodState: APIResponseStatus.Success,
      evaluationPeriod: SerializationUtils.deserializeEvaluationPeriodResponse(
        evaluationPeriods.evaluation_periods[0]
      ),
    },
  };

  it('should look as expected', () => {
    const tree = renderer
      .create(
        <DefaultProviders preloadedState={preloadedState}>
          <EvaluationOrderPage />
        </DefaultProviders>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should behave as expected', () => {
    const { getByText } = renderWithProviders(<EvaluationOrderPage />, {
      preloadedState,
    });
    expect(getByText('Pyyd√§ tarkistusarviointia')).toBeInTheDocument();
  });
});
