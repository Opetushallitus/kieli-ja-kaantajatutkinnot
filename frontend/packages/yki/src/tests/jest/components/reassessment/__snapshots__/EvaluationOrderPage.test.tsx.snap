// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EvaluationOrderPage should look as expected 1`] = `
<div
  className="public-evaluation-order-page MuiBox-root css-0"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column public-evaluation-order-page__grid-container css-1rwmv3x-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item public-evaluation-order-page__grid-container__item-header css-13i4rnv-MuiGrid-root"
    >
      <h1
        className="MuiTypography-root MuiTypography-h1 public-evaluation-order-page__title-heading css-o2w69a-MuiTypography-root"
      >
        Pyydä tarkistusarviointia
      </h1>
      <div
        className="header-separator MuiBox-root css-0"
      />
    </div>
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 public-evaluation-order-page__order-form css-3su884-MuiPaper-root"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        Pyydä tarkistusarviointia seuraavasta tutkinnosta:
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        Tutkinto
        :
         
        <b>
          suomi, keskitaso
        </b>
        <br />
        Testipäivä
        :
         
        <b>
          05/10/2023
        </b>
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        Tähdellä (*) merkityt osiot ovat pakollisia.
      </p>
      <h2
        className="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
      >
        Valitse osakokeet
      </h2>
      <fieldset
        className="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
      >
        <legend
          className="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
        >
          Valitse ne osakokeet, joista haluat pyytää tarkistusarviointia *
        </legend>
        <p
          className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Valitse vähintään yksi osakoe
        </p>
        <div
          className="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
        >
          <label
            className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-wksu0u-MuiFormControlLabel-root"
          >
            <span
              className="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={null}
            >
              <input
                checked={false}
                className="PrivateSwitchBase-input css-1m9pwf3"
                data-indeterminate={false}
                disabled={false}
                onChange={[Function]}
                required={false}
                type="checkbox"
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CheckBoxOutlineBlankIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
            >
              Tekstin ymmärtäminen 100 €
            </span>
          </label>
          <label
            className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-wksu0u-MuiFormControlLabel-root"
          >
            <span
              className="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={null}
            >
              <input
                checked={false}
                className="PrivateSwitchBase-input css-1m9pwf3"
                data-indeterminate={false}
                disabled={false}
                onChange={[Function]}
                required={false}
                type="checkbox"
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CheckBoxOutlineBlankIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
            >
              Puheen ymmärtäminen 100 €
            </span>
          </label>
          <label
            className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-wksu0u-MuiFormControlLabel-root"
          >
            <span
              className="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={null}
            >
              <input
                checked={false}
                className="PrivateSwitchBase-input css-1m9pwf3"
                data-indeterminate={false}
                disabled={false}
                onChange={[Function]}
                required={false}
                type="checkbox"
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CheckBoxOutlineBlankIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
            >
              Kirjoittaminen 100 €
            </span>
          </label>
          <label
            className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-wksu0u-MuiFormControlLabel-root"
          >
            <span
              className="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={null}
            >
              <input
                checked={false}
                className="PrivateSwitchBase-input css-1m9pwf3"
                data-indeterminate={false}
                disabled={false}
                onChange={[Function]}
                required={false}
                type="checkbox"
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CheckBoxOutlineBlankIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
            >
              Puhuminen 100 €
            </span>
          </label>
        </div>
      </fieldset>
      <p
        className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        Tarkistusarvioinnin hinta yhteensä
        : 
        <b>
          0
           €
        </b>
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        Jos osakokeen arviointi muuttuu tarkistusarvioinnissa, palautamme tarkistusarviointimaksun.
      </p>
      <h2
        className="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
      >
        Osallistujan tiedot
      </h2>
      <p
        className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        Täytä alle sen osallistujan tiedot, joka hakee tarkistusarviointia.
      </p>
      <div
        className="public-evaluation-order-page__order-form__participant-details-grid"
      >
        <div
          className="rows"
        >
          <label
            htmlFor="public-evaluation-order-form__field__firstNames"
          >
            <p
              className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              <b>
                Etunimet *
              </b>
            </p>
          </label>
          <div
            className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id="public-evaluation-order-form__field__firstNames"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-ihdtdm"
                >
                  <span
                    className="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="rows"
        >
          <label
            htmlFor="public-evaluation-order-form__field__lastName"
          >
            <p
              className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              <b>
                Sukunimi *
              </b>
            </p>
          </label>
          <div
            className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id="public-evaluation-order-form__field__lastName"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-ihdtdm"
                >
                  <span
                    className="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="rows"
        >
          <label
            htmlFor="public-evaluation-order-form__field__birthdate"
          >
            <p
              className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              <b>
                Syntymäaika *
              </b>
            </p>
          </label>
          <p
            className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Kirjoita muodossa pp.kk.vvvv
          </p>
          <div
            className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id="public-evaluation-order-form__field__birthdate"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-ihdtdm"
                >
                  <span
                    className="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="rows"
        >
          <label
            htmlFor="public-evaluation-order-form__field__email"
          >
            <p
              className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              <b>
                Sähköpostiosoite *
              </b>
            </p>
          </label>
          <p
            className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Esim. nimi@sahkoposti.fi
          </p>
          <div
            className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id="public-evaluation-order-form__field__email"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-ihdtdm"
                >
                  <span
                    className="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <h2
        className="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
      >
        Henkilötietojen käsittelyn ehdot
      </h2>
      <div>
        <p
          className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Noudatamme henkilötietojen käsittelyssä EU:n yleistä tietosuoja-asetusta.
          <br />
          Lue henkilötietojen käsittelyn ehdot Opetushallituksen verkkosivuilta:
        </p>
        <div
          className="columns gapped-xxs"
        >
          <a
            className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
            href="https://www.oph.fi/fi/koulutus-ja-tutkinnot/kieli-ja-kaantajatutkinnot/yleiset-kielitutkinnot-yki/ilmoittautuminen-yki-testiin#anchor-tietoa-henkilotietojen-kasittelysta"
            onBlur={[Function]}
            onFocus={[Function]}
            target="_blank"
          >
            <p
              className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Tietoa henkilötietojen käsittelystä (oph.fi)
            </p>
          </a>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="OpenInNewIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
            />
          </svg>
        </div>
      </div>
      <div
        className="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
      >
        <label
          className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-wksu0u-MuiFormControlLabel-root"
        >
          <span
            className="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <input
              checked={false}
              className="PrivateSwitchBase-input css-1m9pwf3"
              data-indeterminate={false}
              disabled={false}
              onChange={[Function]}
              required={false}
              type="checkbox"
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CheckBoxOutlineBlankIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </span>
          <span
            className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
          >
            Hyväksyn henkilötietojeni käsittelyn *
          </span>
        </label>
      </div>
      <div
        className="public-evaluation-order-page__order-form__action-buttons gapped-xs"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium css-zcbmsk-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          Siirry maksamaan
        </button>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium css-1d3f8j8-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          Keskeytä
        </button>
      </div>
    </div>
  </div>
</div>
`;
